### PR TITLE
Add STFPM model

### PR DESCRIPTION
**Pull request by LukaszMarcjan**
## Summary
Adds the STFPM model. It builds on the shared vision base (from the FastFlow PR)

## What's included
- STFPM model: teacher–student feature-pyramid architecture, loss, and product-aggregated
  anomaly map, as a thin `LightningVisionModel` subclass.
- Config with defaults setting (ResNet-18, 256×256, SGD lr=0.4 / momentum=0.9 / wd=1e-4).
- Concept-incremental example + tests.

### CODE REVIEW

**Comment by Nyderx** [comment]
Rebase to origin/main

File: src/pyclad/vision/models/stfpm/backbones.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    if backbone_name not in _DEFAULT_RETURN_NODES:
+        raise ValueError(
+            f"No default return nodes for '{backbone_name}'. Supported presets: "
+            f"{', '.join(supported_backbone_names())}. Set backbone_return_nodes explicitly to use "
+            "a custom torchvision backbone."
+        )
+    return _DEFAULT_RETURN_NODES[backbone_name]
+
+
+class TorchvisionFeatureExtractor(nn.Module):
```
Comment (Nyderx) [inline src/pyclad/vision/models/stfpm/backbones.py:42]: Isn't it a duplicate of `TorchVisionFeatureExtractor` in src/pyclad/vision/models/fastflow/backbones.py?

**Review by Nyderx — CHANGES_REQUESTED** [review]

**Comment by LukaszMarcjan** [comment]
Rebased onto current `main`
